testserver/testserver.go

```diff
@@ -596,7 +616,12 @@ func NewTestServer(opts ...TestServerOpt) (TestServer, error) {
 	if serverArgs.externalIODir == "" {
 		serverArgs.externalIODir = "disabled"
 	}
-
+	if len(serverArgs.listenAddrHosts) == 0 {
+		serverArgs.listenAddrHosts = make([]string, serverArgs.numNodes)
+		for i := range serverArgs.listenAddrHosts {
+			serverArgs.listenAddrHosts[i] = defaultListenAddrHost
```

Comment:
use `serverArgs.listenAddrHosts[i]` = `serverArgs.listenAddrHost` instead because `defaultListenAddrHost` might have been overriden by `ListenAddrHostOpt`